chartfx-dataset/src/main/java/de/gsi/dataset/serializer/spi/BinarySerialiser.java

```diff
@@ -671,466 +720,457 @@ public static void put(final IoBuffer buffer, final String fieldName, final Enum
         final int nElements = 1;
         final int entrySize = 17; // as an initial estimate
 
-        final long sizeMarkerStart = putArrayHeader(buffer, fieldName, DataType.ENUM, new int[] { 1 },
-                (nElements * entrySize) + 9);
-        final String typeList = Arrays.asList(clazz.getEnumConstants()).stream().map(Object::toString).collect(Collectors.joining(", ", "[", "]"));
-        buffer.putString(clazz.getSimpleName());
-        buffer.putString(enumeration.getClass().getName());
-        buffer.putString(typeList);
-        buffer.putString(enumeration.name());
+        buffer.ensureAdditionalCapacity((nElements * entrySize) + 9);
+        final String typeList = Arrays.stream(clazz.getEnumConstants()).map(Object::toString).collect(Collectors.joining(", ", "[", "]"));
+        buffer.putStringISO8859(clazz.getSimpleName());
+        buffer.putStringISO8859(enumeration.getClass().getName());
+        buffer.putStringISO8859(typeList);
+        buffer.putStringISO8859(enumeration.name());
         buffer.putInt(enumeration.ordinal());
 
-        adjustDataByteSizeBlock(buffer, sizeMarkerStart);
+        updateDataEndMarker(lastFieldHeader);
     }
 
-    public static void put(final IoBuffer buffer, final String fieldName, final float value) {
-        putFieldHeader(buffer, fieldName, DataType.FLOAT);
-        buffer.putFloat(value);
-    }
+    @Override
+    public <K, V, E> void put(final Map<K, V> map, Type keyType, Type valueType, BiFunction<Type, Type[], FieldSerialiser<E>> serialiserLookup) {
```

Comment:
Codacy found an issue: [The method 'put(Map, Type, Type, BiFunction)' has an NPath complexity of 324, current threshold is 200](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5905529)